### PR TITLE
feat: backup rollback

### LINKED ISSUE #330: Support onboarding flow with backup
Support upload flow for onboarding:

- [x] Add link to upload backup screen on onboarding
- [x] Override storage functions with backup file content
- [x] Write mnemonic and password and set them for crypto service
- [x] e2e tests
- [x] Support rollback for unsuccessful uploads

### PR DESCRIPTION
**Pull request by 0xmad**
## Explanation

This PR supports rollback storage for unsuccessful backup uploads.

Details are below:
- [x] Add rollback for unsuccessful backup uploads
- [x] Add restore/download storage methods for backupables
- [x] Add e2e tests for invalid backup uploads

## More Information

Closes #330 

## Screenshots/Screencaps

N/A

## Manual Testing Steps

1. Try to upload any invalid backup file from `packages/e2e/backups`.
2. Check errors
3. Try to upload valid backup file or create account with password
4. Check lock/unlock functions

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

**Review by 0xisk — CHANGES_REQUESTED** [review]
Thanks @0xmad, here are my observations for doing user manual testing: 

**1.** Uploaded `2_invalid_backup.josn`  with having `123456Qq@` as a password for both fields, but got error saying `This ciphertext is not authentic`
Screenshot_20230803_115119

**2.** Trying again with same with having the same upload window opened from step (1), and uploading the same backup file from step (1), and writing the same passwords from step (1). But I got different error this time, "Passwords don't match"
Screenshot_20230803_115038

**3.** Then still with having the same window opened, I tried the same INVALID_BACKUP files. But was always getting `Passwords don't match`
Screenshot_20230803_115131
Screenshot_20230803_115137
Screenshot_20230803_115147

**4.** I closed the same window opened for Steps (1, 2, 3) and clicked again on Upload backup from settings page to open a new window, then I tried with 0_backup file which is simply the valid file and as expected it worked without errors.

**Comment by 0xisk** [comment]
@0xmad I noticed that you checked `IF this PR fixes a bug, a test that would have caught the bug has been added` but it is not clear to me which bug it resolves, also there is no issue for this attached.

**Comment by 0xmad** [comment]
@0xisk thanks, remove logout after unsuccessful upload.
Regarding the bug: it fixes bugs related to unsuccessful uploads. Tests are added to e2e folder.

**Comment by 0xisk** [comment]
> @0xisk thanks, remove logout after unsuccessful upload. Regarding the bug: it fixes bugs related to unsuccessful uploads. Tests are added to e2e folder.

For all the invalid backup files it shows `This ciphertext is not authentic`, I think that is a vague response to the user, I'd say rather we could print in this case `This backup file is not authentic`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -41,6 +41,8 @@ How should reviewers and QA manually test your changes? For instance:
 ## Pre-Merge Checklist
 
 - [ ] PR template is filled out
+- [ ] Pre-commit and pre-push hook checks are passed
+- [ ] e2e tests are passed locally
```
Comment (0xisk) [inline .github/PULL_REQUEST_TEMPLATE.md:45]:
```suggestion
- [ ] E2E tests are passed locally
```

**Review by 0xisk — CHANGES_REQUESTED** [review]

**Comment by 0xmad** [comment]
> > @0xisk thanks, remove logout after unsuccessful upload. Regarding the bug: it fixes bugs related to unsuccessful uploads. Tests are added to e2e folder.
> 
> For all the invalid backup files it shows `This ciphertext is not authentic`, I think that is a vague response to the user, I'd say rather we could print in this case `This backup file is not authentic`

It's not possible to distinguish these errors and this is not a common case for user.

**Comment by 0xisk** [comment]
> > > @0xisk thanks, remove logout after unsuccessful upload. Regarding the bug: it fixes bugs related to unsuccessful uploads. Tests are added to e2e folder.
> > 
> > 
> > For all the invalid backup files it shows `This ciphertext is not authentic`, I think that is a vague response to the user, I'd say rather we could print in this case `This backup file is not authentic`
> 
> It's not possible to distinguish these errors and this is not a common case for user.

Actually `getAuthenticCiphertext()` which throw this error, only used through `uploadEncryptedStorage()` functions in all backup supported services. So I think we can just rename this function to be `getAuthenticBackup()` and the error would be `This Backup file is not authentic`

**Review by 0xisk — APPROVED** [review]
Approved, thanks @0xmad!

**Pull request merged**